packages/agentmark-core/src/template_engines/templatedx.ts

```
@@ -25,8 +27,17 @@ type ExtractedField = {
 type SharedContext = {
   "__agentmark-extractTextPromises"?: Promise<ExtractedField>[];
 };
+export enum Role {
+  user = "user",
+  system = "system",
+  assistant = "assistant",
+}
 
 const USER = "User";
+const SYSTEM = "System";
+const ASSISTANT = "Assistant";
+const SPEECH_PROMPT = "SpeechPrompt";
+const IMAGE_PROMPT = "ImagePrompt";
```

Those are the prompt tags that users should use for speech and image generation from now on instead of relying on `msgs`. If they add any messgaes in the mdx file, then they won't be sent to the AI model, should we throw an error instead of ignoring the role tags?